lib/ansible/modules/source_control/git.py

```diff
@@ -432,8 +448,25 @@ def clone(git_path, module, repo, dest, remote, depth, version, bare,
                         "HEAD, branches, tags or in combination with refspec.")
     if reference:
         cmd.extend(['--reference', str(reference)])
+    separate_git_dir_fallback = False
+
+    if separate_git_dir:
+        if os.path.exists(separate_git_dir):
+            module.fail_json(msg='Separate-git-dir path %s already exists.' % separate_git_dir)
+        git_version_used = git_version(git_path, module)
+        if git_version_used is None:
```

Comment:
Instead of lots of if/else, you could follow EAFP principle and do:
```python
try:
    if git_version(git_path, module) < LooseVersion('1.7.5'):
        separate_git_dir_fallback = True
    else:
        cmd.append('--separate-git-dir=%s' % separate_git_dir)
except TypeError:  # will happen when comparing `None`
    module.fail_json(msg="Cannot find git executable at '%s'" % git_path)
```